packages/artifact/src/internal/upload/upload-artifact.ts

```diff
@@ -77,30 +83,50 @@ export async function uploadArtifact(
   }
 
   // Upload zip to blob storage
-  const uploadResult = await uploadZipToBlobStorage(createArtifactResp.signedUploadUrl, zipUploadStream)
+  const uploadResult = await uploadZipToBlobStorage(
+    createArtifactResp.signedUploadUrl,
+    zipUploadStream
+  )
   if (uploadResult.isSuccess === false) {
     return {
       success: false
     }
   }
 
   // finalize the artifact
-  const finalizeArtifactResp = await artifactClient.FinalizeArtifact({
+  const finalizeArtifactReq: FinalizeArtifactRequest = {
     workflowRunBackendId: backendIds.workflowRunBackendId,
     workflowJobRunBackendId: backendIds.workflowJobRunBackendId,
     name: name,
     size: uploadResult.uploadSize!.toString()
-  })
+  }
+
+  if (uploadResult.md5Hash) {
+    finalizeArtifactReq.hash = StringValue.create({
+      value: `md5: ${uploadResult.md5Hash!}`
```

Comment:
very nit: remove the space
```suggestion
      value: `md5:${uploadResult.md5Hash!}`
```